### ISSUE TITLE
Memory leak while continuous prediction

### ISSUE THREAD

**Issue by silvernox-dev** [matterport/Mask_RCNN#2372]
Am hosting a mask rcnn model on flask app, the predictions work ok for a couple of inference but then after sometime I get CUDA OOM error -
`(0) Resource exhausted: OOM when allocating tensor with shape[19,512,256,256] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc`

To reproduce the error - 
```
while True:
    print('Predicting')

    with graph.as_default():
        with sess.as_default():
            res = model.detect(all_images, verbose=0)
            r2 = res[0]
```

Am using -
tensorflow-gpu==1.15.2
Keras==2.2.4
